### ISSUE TITLE
UI: proposed new "Preferences" window

### ISSUE THREAD

**Issue by amivaleo** [Guake/guake#1618]
Hi!

I made this for fun: [prefs.zip](https://github.com/Guake/guake/files/3317773/prefs.zip)

Consider that more as a sketch for a mockup more than an replacement for the current window.

I hope you will like it and that it can give you some idea for the gui. Feel free to use that file or part of it at any extent. :)

**Comment by gsemet** [comment]
Do you have a screenshot ?

I am open to improvement on the pref panels, it is always tricky to maintain it and have it working on any screen size :(

**Comment by amivaleo** [comment]
Sure!

![Schermata da 2019-06-28 22-06-48](https://user-images.githubusercontent.com/24316547/60365403-4aa4e780-99f1-11e9-9212-b1be3163b3c9.png)

![Schermata da 2019-06-28 22-06-51](https://user-images.githubusercontent.com/24316547/60365459-70ca8780-99f1-11e9-95cc-03e2885f8fd2.png)

![Schermata da 2019-06-28 22-06-55](https://user-images.githubusercontent.com/24316547/60365477-7a53ef80-99f1-11e9-8aaa-4ce70987242c.png)

But I truly suggest you to download the file and test it on glade, so to have a deep understanding of the idea I came up with. :)

**Comment by amivaleo** [comment]
So... No comment? :(

**Comment by gsemet** [comment]
Hi ! I was quite busy out of Guake, so I just had a quick look at it.

My opinion is this a pretty good. It’s more modern and elegant :) If you prepare a pull request I will definitively merge it. I have no time to work on properly integrate it myself, however :(

Just a few question: on small screen, does this display a scrollbar for long list of option?

**Comment by Davidy22** [comment]
If you can already work with glade, you should be fine: I'm looking at glade 3.22 and there's an ID field at the top of the general tab that has a value for every option widget in our current preferences file, and is blank in the attached preferences window. If you modify your IDs to match up with our current IDs, your preferences window should hook into our code without you having to actually touch code.

**Comment by amivaleo** [comment]
> If you can already work with glade, you should be fine: I'm looking at glade 3.22 and there's an ID field at the top of the general tab that has a value for every option widget in our current preferences file, and is blank in the attached preferences window. If you modify your IDs to match up with our current IDs, your preferences window should hook into our code without you having to actually touch code.

I suppose that might work IF I used the very same widget, which is not the case since I use switches instead of checkboxes.

Even without a way to test it, I'm prone to wonder if filling the ID fields wouldn't be only a mere waste of time. :\

**Comment by Davidy22** [comment]
Well if you're fully unsure about being able to link correctly, you can also just update the preferences menu to match master, put the glade file up and hope someone else fills it in, or shows you how to fill it in with an example.

**Comment by amivaleo** [comment]
> Well if you're fully unsure about being able to link correctly, you can also just update the preferences menu to match master, put the glade file up

Are you talking about a pull request?

> and hope someone else fills it in, or shows you how to fill it in with an example.

Would you do that? I'd be glad to learn by examples. I don't have much time, I can't start from reading all the required "literature". But if someone would like to teach me a couple of very practical things or so, I'd be glad to help with guake and not only with that.

**Comment by Davidy22** [comment]
Yeah, it would be a pull request where you replace the glade file with your own. If you put one up I can take a shot at hooking some buttons to code to see what has to be done and I can fill you in on what I figure out.

**Comment by amivaleo** [comment]
> Yeah, it would be a pull request where you replace the glade file with your own. If you put one up I can take a shot at hooking some buttons to code to see what has to be done and I can fill you in on what I figure out.

Thank you!

I tagged you on the pull request.
